R/derive_params_growth_height.R

```diff
@@ -0,0 +1,284 @@
+#' A numeric vector is expected, e.g. `AVAL`, `VSSTRESN`
+#'
+#' @param set_values_to_sds Variables to be set for Z-Scores
+#'
+#'  The specified variables are set to the specified values for the new
+#'  observations. For example,
+#'   `set_values_to_sds(exprs(PARAMCD = “BMIASDS”, PARAM = “BMI-for-height z-score”))`
+#'  defines the parameter code and parameter.
+#'
+#' *Permitted Values*: List of variable-value pairs
+#'
+#'  If left as default value, `NULL`, then parameter not derived in output dataset
+#'
+#' @param set_values_to_pctl Variables to be set for Percentile
+#'
+#'  The specified variables are set to the specified values for the new
+#'  observations. For example,
+#'   `set_values_to_pctl(exprs(PARAMCD = “BMIAPCTL”, PARAM = “BMI-for-height percentile”))`
+#'  defines the parameter code and parameter.
+#'
+#' *Permitted Values*: List of variable-value pair
+#'
+#'  If left as default value, `NULL`, then parameter not derived in output dataset
+#'
+#' @return The input dataset additional records with the new parameter added.
+#'
+#'
+#' @family der_prm_bds_vs
+#'
+#' @keywords der_prm_bds_vs
+#'
+#' @export
+#'
+#' @examples
+#' library(dplyr)
+#' library(lubridate)
+#' library(rlang)
+#' library(admiral)
+#'
+#' advs <- dm_peds %>%
+#'   select(USUBJID, BRTHDTC, SEX) %>%
+#'   right_join(., vs_peds, by = "USUBJID") %>%
+#'   filter(USUBJID != "PEDS-1010") %>%
+#'   mutate(
+#'     VSDT = ymd(VSDTC),
+#'     BRTHDT = ymd(BRTHDTC)
+#'   ) %>%
+#'   derive_vars_duration(
+#'     new_var = AGECUR_M,
+#'     new_var_unit = CURU_M,
+#'     start_date = BRTHDT,
+#'     end_date = VSDT,
+#'     out_unit = "months",
+#'     trunc_out = FALSE
+#'   )
+#'
+#' heights <- vs_peds %>%
+#'   filter(VSTESTCD == "HEIGHT") %>%
+#'   select(USUBJID, VSSTRESN, VSSTRESU, VSDTC) %>%
+#'   rename(
+#'     HEIGHT = VSSTRESN,
+#'     HEIGHTU = VSSTRESU
+#'   )
+#'
+#' advs <- advs %>%
+#'   right_join(., heights, by = c("USUBJID", "VSDTC"))
+#'
+#' advs_under2 <- advs %>%
+#'   filter(AGECUR_M <= 2)
+#'
+#' advs_over2 <- advs %>%
+#'   filter(AGECUR_M > 2)
+#'
+#' who_under2 <- bind_rows(
+#'   (admiralpeds::who_wt_for_lgth_boys %>%
+#'     mutate(
+#'       SEX = "M",
+#'       height_unit = "cm"
+#'     )
+#'   ),
+#'   (admiralpeds::who_wt_for_lgth_girls %>%
+#'     mutate(
+#'       SEX = "F",
+#'       height_unit = "cm"
+#'     )
+#'   )
+#' ) %>%
+#'   rename(
+#'     HEIGHT_LENGTH = Length,
+#'     HEIGHT_LENGTHU = height_unit
+#'   )
+#'
+#' who_over2 <- bind_rows(
+#'   (admiralpeds::who_wt_for_ht_boys %>%
+#'     mutate(
+#'       SEX = "M",
+#'       height_unit = "cm"
+#'     )
+#'   ),
+#'   (admiralpeds::who_wt_for_ht_girls %>%
+#'     mutate(
+#'       SEX = "F",
+#'       height_unit = "cm"
+#'     )
+#'   )
+#' ) %>%
+#'   rename(
+#'     HEIGHT_LENGTH = Height,
+#'     HEIGHT_LENGTHU = height_unit
+#'   )
+#'
+#'
+#' derive_params_growth_height(
+#'   advs_under2,
+#'   sex = SEX,
+#'   height = HEIGHT,
+#'   height_unit = HEIGHTU,
```

Comment:
same comment above to match template - think it was HGTTMP from memory? which is better as it doesn't then confuse with the height PARAMCD in VS, as hard to follow otherwise that this one is a variable being input not a parameter